src/js/components/Layer/__tests__/Layer-test.js

```diff
@@ -212,4 +212,34 @@ describe('Layer', () => {
     ref.current.componentDidMount();
     expect(queryByTestId(document, 'test-layer-container')).toMatchSnapshot();
   });
+
+  test('focus on layer', () => {
+    render(
+      <Grommet>
+        <Layer data-testid="focus-layer-test">
+          <input />
+        </Layer>
+      </Grommet>,
+    );
+
+    const layerNode = getByTestId(document, 'focus-layer-test');
+    expect(layerNode).toMatchSnapshot();
+    expect(document.activeElement.nodeName).toBe('A');
+  });
+
+  test('not steal focus from an autofocus focusable element', () => {
+    /* eslint-disable jsx-a11y/no-autofocus */
+    render(
+      <Grommet>
+        <Layer data-testid="focus-input-test">
+          <input autoFocus />
+          <button type="button">Button</button>
+        </Layer>
+      </Grommet>,
+    );
+    /* eslint-disable jsx-a11y/no-autofocus */
+    const layerNode = getByTestId(document, 'focus-input-test');
+    expect(layerNode).toMatchSnapshot();
+    expect(document.activeElement.nodeName).toBe('INPUT');
```

Comment:
It would be tighter to give the input a `data-testid` and validate against that.

Reply:
Much better. At time of writing I realized there could be other possible ways of identifying this element better, so I made it more specific with a `datat-testid`.